src/context/AuthProvider/index.tsx

```diff
@@ -0,0 +1,16 @@
+import React, { createContext, useReducer } from 'react';
+
+import { authReducer, authState } from './reducer';
+
+export const AuthContext = createContext({});
+
+interface Props {
+  children: React.ReactNode;
+}
+
+function AuthProvider({ children }: Props) {
+  const [authStore, dispatch] = useReducer(authReducer, authState);
+  return <AuthContext.Provider value={[authStore, dispatch]}>{children}</AuthContext.Provider>;
+}
```

Comment:
Same. You could define the HOC I wrote below a re use for both the auth and global provider or just write each case where it belongs.